pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -160,6 +160,30 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 	}
 	s.End(true)
 
+	log.Info("\nUpdating services")
+	// fetch the "kubernetes inlined components" to create them on cluster
+	// from odo standpoint, these components contain yaml manifest of an odo service or an odo link
+	k8sComponents, err := a.Devfile.Data.GetComponents(parsercommon.DevfileOptions{
+		ComponentOptions: parsercommon.ComponentOptions{ComponentType: devfilev1.KubernetesComponentType},
+	})
+	if err != nil {
+		return errors.Wrap(err, "error while trying to fetch service(s) from devfile")
+	}
+	labels := componentlabels.GetLabels(a.ComponentName, a.AppName, true)
+	// create the Kubernetes objects from the manifest and delete the ones not in the devfile
+	needRestart, err := service.PushServiceFromKubernetesInlineComponents(a.Client.GetKubeClient(), k8sComponents, labels)
+	if err != nil {
+		return errors.Wrap(err, "failed to create service(s) associated with the component")
+	}
+
+	if componentExists && needRestart {
+		time.Sleep(time.Second)
```

Comment:
Why are we adding a sleep here?

Reply:
To give a delay for the SB operator to update the deployment. Without this delay, the following `WaitForDeploymentRollout` does not wait for the pod to restart.
It could be possible to find a better way, probably by waiting for the current pod to be in non running state.

Reply:
This seems like a hack which is unreliable. We have generally avoided adding sleep into the code base. We should find a better way, maybe by looking into the status of the service binding resource etc.

Reply:
I agree, I'll try to find a better way

Reply:
Thanks :+1:

Reply:
I made a change to wait for the pod to be in a non ready state (or to de deleted).